core/market/decentralized/src/matcher/matcher.rs

```diff
@@ -66,30 +68,34 @@ pub struct EventsListeners {
 }
 
 /// Responsible for storing Offers and matching them with demands.
+///
+/// Note: Should be lightweight cloneable object. If you
+#[derive(Clone)]
 pub struct Matcher {
```

Comment:
I feel like this is rather OfferStore than a Matcher, there a lot of db manipulations, and just small fraction will be actual matching (now there is only a TODO, so 100% of code is OfferStore)
```suggestion
pub struct OfferStore {
```

Reply:
Where are you going to implement matching?

Reply:
Note that we still need single object that exposes interface for `Market`.

I don't want to change this now... It doesn't matter, what this struct does now, but what it will do in future. If it won't fit the new code structure, we will change this name, but at this moment we don't now exactly.